src/fidgets/ui/channel.tsx

```
@@ -0,0 +1,276 @@
+        )}
+      </div>
+      {(description || externalLink?.url) && (
+        <div className="space-y-2">
+          {description && (
+            <p className="text-sm leading-relaxed text-slate-700 dark:text-slate-200">
+              {description}
+            </p>
+          )}
+          {externalLink?.url && (() => {
+            const isValid = isValidHttpUrl(externalLink.url);
+            const safeUrl = useSafeUrl(externalLink.url);
```

Biome (2.1.2)

[error] This hook is being called from a nested function, but all hooks must be called unconditionally from the top-level component.

For React to preserve state between calls, hooks needs to be called unconditionally and always in the same order.
See https://reactjs.org/docs/hooks-rules.html#only-call-hooks-at-the-top-level

(lint/correctness/useHookAtTopLevel)